client/data/site-settings/with-site-settings.js

```diff
@@ -0,0 +1,151 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { useCallback, useState } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import useSiteSettingsQuery from 'calypso/data/site-settings/use-site-settings-query';
+import useUpdateSiteSettingsMutation from 'calypso/data/site-settings/use-update-site-settings-mutation';
+import { useProtectForm } from 'calypso/lib/protect-form';
+import { recordGoogleEvent, recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import getCurrentRouteParameterized from 'calypso/state/selectors/get-current-route-parameterized';
+
+const noticeOptions = {
+	duration: 10000,
+	id: 'site-settings-save',
+};
+
+const withSiteSettings = createHigherOrderComponent( ( Wrapped ) => {
+	const WithSiteSettings = ( props ) => {
+		const { siteId } = props;
+		const [ unsavedSettings, setUnsavedSettings ] = useState( {} );
```

Comment:
Nit: this creates a new object on every rerender. You might want to use [lazy init](https://kentcdodds.com/blog/use-state-lazy-initialization-and-function-updates):

```suggestion
		const [ unsavedSettings, setUnsavedSettings ] = useState( () => {} );
```